### PR TITLE
lib/protoparser/opentelemetry/firehose: escape requestID before returning it to user

### PR DESCRIPTION
**Pull request by hagen1778**
All user input should be sanitized before rendering. This should prevent possible attacks. See https://github.com/VictoriaMetrics/VictoriaMetrics/security/code-scanning/203

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### CODE REVIEW

**Review by dima-vm — APPROVED** [review]

File: lib/protoparser/opentelemetry/firehose/http.go [hunk 1]
```diff
@@ -12,11 +13,12 @@ import (
...
 	requestID := r.Header.Get("X-Amz-Firehose-Request-Id")
 	if requestID == "" {
-		// This isn't a AWS firehose request - just return an empty response in this case.
+		// This isn't an AWS firehose request - just return an empty response in this case.
 		w.WriteHeader(http.StatusOK)
 		return
 	}
 
+	requestID = html.EscapeString(requestID)
 	body := fmt.Sprintf(`{"requestId":%q,"timestamp":%d}`, requestID, time.Now().UnixMilli())
```
Comment (dima-vm) [inline lib/protoparser/opentelemetry/firehose/http.go:22]: I'd name "timestamp" -> "timestamp_ms" if possible.

Comment (AndrewChubatiuk) [reply lib/protoparser/opentelemetry/firehose/http.go:22]: we cannot rename this field, as it's required by AWS https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html#responseformat

**Review by AndrewChubatiuk — APPROVED** [review]
LGTM

**Comment by valyala** [comment]
@hagen1778 , @dima-vm , @AndrewChubatiuk , this pull request has no sense, so it has been reverted in 7a3394bbe189c16ead083da721a9f1bff5473dff . Please read the description of the revert commit for details.

**Pull request merged**